src/DependencyManagement/DependencyManager.cs

```diff
@@ -0,0 +1,372 @@
+                throw dependencyInstallationException;
+            }
+        }
+
+        #region Helper_Methods
+
+        /// <summary>
+        /// Installs managed dependencies modules specified in Requirements.psd1 for the function app.
+        /// </summary>
+        private void InstallManagedDependencyModule(PowerShell pwsh, ILogger logger)
+        {
+            if (Dependencies.Count == 0)
+            {
+                // If there are no dependencies to install, log and return.
+                logger.Log(LogLevel.Trace, PowerShellWorkerStrings.FunctionAppDoesNotHaveDependentModulesToInstall, isUserLog: true);
+                return;
+            }
+
+            if (!_shouldUpdateFunctionAppDependencies)
+            {
+                // The function app already has the latest dependencies installed.
+                logger.Log(LogLevel.Trace, PowerShellWorkerStrings.LatestFunctionAppDependenciesAlreadyInstalled, isUserLog: true);
+                return;
+            }
+
+            // Install the function dependencies.
+            logger.Log(LogLevel.Trace, PowerShellWorkerStrings.InstallingFunctionAppDependentModules, isUserLog: true);
+
+            if (Directory.Exists(DependenciesPath))
+            {
+                // Save-Module supports downloading side-by-size module versions. However, we only want to keep one version at the time.
+                // If the ManagedDependencies folder exits, remove all its contents.
+                DependencyManagementUtils.EmptyDirectory(DependenciesPath);
+            }
+            else
+            {
+                // If the destination path does not exist, create it.
+                Directory.CreateDirectory(DependenciesPath);
+            }
+
+            try
+            {
+                foreach (var module in Dependencies)
+                {
+                    var moduleName = module.Name;
+                    var latestVersion = module.LatestVersion;
+
+                    // Save the module to the given path
+                    pwsh.AddCommand("PowerShellGet\\Save-Module")
+                        .AddParameter("Repository", Repository)
+                        .AddParameter("Name", moduleName)
+                        .AddParameter("RequiredVersion", latestVersion)
+                        .AddParameter("Path", DependenciesPath)
+                        .AddParameter("Force", true)
+                        .AddParameter("ErrorAction", "Stop")
+                        .InvokeAndClearCommands();
+
+                    var message = string.Format(PowerShellWorkerStrings.ModuleHasBeenInstalled, moduleName, latestVersion);
+                    logger.Log(LogLevel.Trace, message, isUserLog: true);
+                }
+            }
+            finally
+            {
+                // Clean up
+                pwsh.AddCommand("Microsoft.PowerShell.Core\\Remove-Module")
```

Comment:
I'm curious how long this takes... it might take too long and we should just keep the modules around.

Reply:
Removing the module was a request from @daxian-dbw . I also think we should remove any modules that were not imported directly by the user. 

Here is how long it takes to remove these two modules using measure-command.
```
PS C:\> measure-command {Remove-Module -name powershellget, packagemanagement}


Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 0
Milliseconds      : 56
Ticks             : 566368
TotalDays         : 6.55518518518518E-07
TotalHours        : 1.57324444444444E-05
TotalMinutes      : 0.000943946666666667
TotalSeconds      : 0.0566368
TotalMilliseconds : 56.6368

```

Reply:
Is that how long it took in the Functions sand box with 1 core? Or on your multi-core local machine.

I'm not blocking on this... but if we're looking to squeeze out some cold start time, this might be a good place to do it. @daxian-dbw

Reply:
This whole dependency loading logic is hooked up with `PowerShellManager Pool/PowerShell Manager` as the extra `initAction`. We are doing extra initialization in the Runspace that is to be used for user functions, so I think the good practice is for the `initAction` to clean up after its init work is done. For this particular case, maybe it's not a big deal to keep those two modules around, but in general it's better to clean them up.

Reply:
Exactly. Leave it in for now, but let the record know that we can optimize slightly by not cleaning up these two modules

Reply:
Sounds good.